template/python3/Dockerfile

```diff
@@ -1,56 +1,88 @@
-FROM --platform=${TARGETPLATFORM:-linux/amd64} ghcr.io/openfaas/classic-watchdog:0.2.0 as watchdog
-FROM --platform=${TARGETPLATFORM:-linux/amd64} python:3-alpine
-
-ARG TARGETPLATFORM
-ARG BUILDPLATFORM
+# Builder stage that allows you to use git modules from private repos
+FROM --platform=${TARGETPLATFORM:-linux/amd64} python:3-alpine as builder
 
-# Allows you to add additional packages via build-arg
+# Basic user, python and certificate setup
 ARG ADDITIONAL_PACKAGE
-
-COPY --from=watchdog /fwatchdog /usr/bin/fwatchdog
-RUN chmod +x /usr/bin/fwatchdog
 RUN apk --no-cache add ca-certificates ${ADDITIONAL_PACKAGE}
+RUN addgroup -S app && adduser app -S -G app
+WORKDIR /home/app/
+RUN chown -R app /home/app && \
+    mkdir -p /home/app/python && chown -R app /home/app
+USER app
+ENV PATH=$PATH:/home/app/.local/bin:/home/app/python/bin/
+ENV PYTHONPATH=$PYTHONPATH:/home/app/python
 
+# Token to be provided as argument
+ARG GIT_TOKEN=no_token_set
 
-# Add non root user
-RUN addgroup -S app && adduser app -S -G app
+# Install git and make the git token available as environment variable
+USER root
+RUN apk --no-cache add git
+ENV GIT_TOKEN=${GIT_TOKEN}
```

Comment:
It may be better to avoid leaking the ENV into the build, and instead prefixing the value in the pip install command.

Reply:
I was going to say the same thing, like this

```
RUN GIT_TOKEN=${GIT_TOKEN} pip install -r requirements.txt --target=/home/app/python
```

Reply:
I didn't think that the env var would be leaked if used in the builder stage, but I do agree that it's neater the way that @LucasRoesler proposes. That way it won't unnecessarily be available to all the other commands.

Reply:
Changed